browser/js/global.js

```diff
@@ -1240,16 +1261,27 @@ function getInitializerClass() {
 				else
 					data = this.doSlice(arr, i, i + size);
 
-				if (serial !== that.inSerial + 1) {
-					global.app.console.debug('Error: serial mismatch ' + serial + ' vs. ' + (that.inSerial + 1));
-				}
-				that.inSerial = serial;
-				try {
-					this.onmessage({ data: data });
-				} catch (e) {
-					global.app.console.error(e);
-					global.app.console.warn(`Failed processing a ProxySocket message (due to ${e}), ignoring`);
-					// It's better to ignore any failures rather than to lose the rest of the messages in this packet
+				if (serial === that.inSerial + 1) {
```

Comment:
Looks reasonable - but I would share this code - we should just push all the messages onto the queue and then call this.processBufferedMessage() to process them rather than duplicating code =)

Thanks!